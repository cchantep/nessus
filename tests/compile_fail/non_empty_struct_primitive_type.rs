// This test verifies that primitive types do NOT implement NonEmptyStruct

use nessus::NonEmptyStruct;

fn main() {
    // This should fail to compile because String does not implement NonEmptyStruct
    fn _assert_implements_non_empty_struct<T: NonEmptyStruct>() {}
    _assert_implements_non_empty_struct::<String>();
}
