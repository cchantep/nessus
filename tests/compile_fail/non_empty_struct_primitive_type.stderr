error[E0277]: the trait bound `String: NonEmptyStruct` is not satisfied
 --> tests/compile_fail/non_empty_struct_primitive_type.rs:8:43
  |
8 |     _assert_implements_non_empty_struct::<String>();
  |                                           ^^^^^^ the trait `NonEmptyStruct` is not implemented for `String`
  |
note: required by a bound in `_assert_implements_non_empty_struct`
 --> tests/compile_fail/non_empty_struct_primitive_type.rs:7:47
  |
7 |     fn _assert_implements_non_empty_struct<T: NonEmptyStruct>() {}
  |                                               ^^^^^^^^^^^^^^ required by this bound in `_assert_implements_non_empty_struct`
